### PR TITLE
HRIS-124 [Markup] Modify File Leave/Undertime Request Modal

### PR DESCRIPTION
**Pull request by jsvelte**
## Issue Link

- https://framgiaph.backlog.com/view/HRIS-124

## Definition of Done

- [x] Modified Newly Updated File Leave/Undertime Modal
- [x] Applied React Hook Form
- [x] Removed existing integration 
- [x] All default select are now changed into react-select package 
- [x] Added Reset button to reset field errors and empty the fields
- [x] Created constant data && Stretched Manager Overtime Table

## Notes
- All the field of select are now in react-select package
- https://www.figma.com/file/w3lR4Elj1q5yZgK3Xw0eIx/HRIS-Wireframe?node-id=3106%3A20695&t=N9fQqjoGoDCYIAVT-4
- https://www.figma.com/file/w3lR4Elj1q5yZgK3Xw0eIx/HRIS-Wireframe?node-id=3106%3A20523&t=N9fQqjoGoDCYIAVT-4

## Pre-condition
cd client && npm run dev
cd api && dotnet run watch

## Expected Output

It should now have the newly updated markup for filing Leave/Undertime Request Modal
Applied react-hook-form

## Screenshots/Recordings
[modified.webm](https://user-images.githubusercontent.com/108642414/219319994-d29ae35e-5c77-431f-b1c4-e5eb0805a1b2.webm)

### CODE REVIEW

**Review by arden-dave — APPROVED** [review]
Approve. 👍

File: client/src/utils/constants/dummyAddNewLeaveFields.ts [hunk 1]
```diff
@@ -84,63 +84,74 @@ export const projectLeaders = [
...
   {
-    id: 3,
+    label: 'Vacation leave',
     value: 'Vacation leave'
   },
   {
-    id: 4,
+    label: 'Maternity/Paternity leave',
     value: 'Maternity/Paternity leave'
   }
```
Comment (abduljalilpalala) [inline client/src/utils/constants/dummyAddNewLeaveFields.ts:103]: Create constants for each label.
Sample: SICK_LEAVE, BEREAVEMENT_LEAVE, VACATION_LEAVE, MATERNITY_PATERNITY_LEAVE

File: client/src/utils/constants/dummyAddNewLeaveFields.ts [hunk 2]
```diff
@@ -84,63 +84,74 @@ export const projectLeaders = [
...
   {
-    id: 3,
+    label: '0.69 = 5.5hrs (afternoon only)',
     value: '0.69 = 5.5hrs (afternoon only)'
   },
   {
-    id: 4,
+    label: '1 = 8 (whole day)',
     value: '1 = 8 (whole day)'
   }
```
Comment (abduljalilpalala) [inline client/src/utils/constants/dummyAddNewLeaveFields.ts:122]: Create constants for each label. (Refer to previous comment)

File: client/src/utils/constants/dummyAddNewLeaveFields.ts [hunk 3]
```diff
@@ -84,63 +84,74 @@ export const projectLeaders = [
...
   {
-    id: 4,
+    label: '0.31 = 2.5hrs',
     value: '0.31 = 2.5hrs'
   },
   {
-    id: 5,
+    label: '0.38 = 3hrs',
     value: '0.38 = 3hrs'
   }
```
Comment (abduljalilpalala) [inline client/src/utils/constants/dummyAddNewLeaveFields.ts:145]: Create constants for each label. (Refer to previous comment)

**Review by abduljalilpalala — CHANGES_REQUESTED** [review]
Please check comments 💬

**Review by abduljalilpalala — APPROVED** [review]
### Approved! ✅

**Pull request merged**